graylog2-server/src/main/java/org/graylog2/inputs/converters/DateConverter.java

```diff
@@ -39,20 +48,37 @@ public DateConverter(Map<String, Object> config) throws ConfigurationException {
             throw new ConfigurationException("Missing config [date_format].");
         }
 
-        dateFormat = ((String) config.get("date_format")).trim();
+        this.dateFormat = ((String) config.get("date_format")).trim();
+        this.timeZone = buildTimeZone(config.get("time_zone"));
+    }
+
+    private static DateTimeZone buildTimeZone(Object timeZoneId) {
```

Comment:
You're right. I've opted to extract the common functionality into an abstract base class.